bootloader/lk.go

```diff
@@ -170,7 +425,17 @@ func (l *lk) ExtractKernelAssets(s snap.PlaceInfo, snapf snap.Container) error {
 			return fmt.Errorf("cannot open unpacked %s: %v", bootImg, err)
 		}
 		defer bif.Close()
-		bpart := filepath.Join(l.dir(), bootPartition)
+		var bpart string
+		// TODO: for RoleSole bootloaders this will eventually be the same
+		// codepath as for non-RoleSole bootloader
+		if l.role == RoleSole {
+			bpart = filepath.Join(l.dir(), bootPartition)
+		} else {
+			bpart, _, err = l.envFileForPartName(bootPartition)
```

Comment:
this feels an abuse of something called envFileForPatName

Reply:
renamed